demos/dotnet/dotnet/dotnet.cs

```diff
@@ -240,6 +264,29 @@ public static ProjectProperties InitializeProperties(string[] args, Log log, boo
         // CSC OPTIONS
         properties.CscOptions.Add("/nostdlib");
         properties.CscOptions.Add("/noconfig");
+        if (Array.Exists(args, element => element == "/unsafe")) 
+        {
+            properties.CscOptions.Add("/unsafe");
+        }
+
+        if (Array.Exists(args, element => element == "/optimize"))
+        {
+            properties.CscOptions.Add("/optimize");
+        }
+
+        var platformOption = Array.Find(args, element => element.StartsWith("/platform"));
+
+        if (platformOption != null)
+        {
+            if (!(platformOption == "/platform:anycpu32bitpreferred" && buildDll)) properties.CscOptions.Add(platformOption);
```

Comment:
The other options this PR add are simple options that need to be just passed to the compiler. But the architecture option is not so simple. In particular, the routine that copies package files to the output folder needs to take architecture into the account. 

BTW, feel free to remove either fix this issue, or remove /platfrom from this PR and do it as a separate work item.